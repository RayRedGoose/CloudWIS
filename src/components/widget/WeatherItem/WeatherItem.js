import React from 'react'
import './WeatherItem.scss'

const WeatherItem = (props) => {
  return (
    <section className={`weather-item weather-item__${props.itemType}`}>
      <p className={`header-${props.blockType}`}>{ props.weekDay }</p>
      <p className={ "number " + props.typeAbbr }> { props.number }</p>
      <p>{ props.itemType }</p>
    </section>
  )
}

export default WeatherItem
