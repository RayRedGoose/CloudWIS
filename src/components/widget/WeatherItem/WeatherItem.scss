.weather-item {
  background-color: #FFFFFF;
  border-radius: 20px;
  text-align: center;
  padding: 20px;
  margin-right: 10px;
  min-width: 125px;
  p {
    text-transform: uppercase;
    &:first-child {
      padding-bottom: 5px;
      border-bottom: 1px solid #000;
      margin-bottom: 10px;
    }
  }
}

.number {
  font-size: 48px;
  margin-bottom: 10px;
  color: #0D324D;
  &:after {
    font-size: 28px;
    font-weight: 400;
  }
}

.temp {
  &:after {
    content: "\00B0 F";
  }
}

.wind {
  &:after {
    content: "mph";
    text-transform: lowercase;
  }
}

.humidity {
  &:after {
    content: "%";
    text-transform: lowercase;
  }
}

.visibility {
  &:after {
    content: "mi";
    text-transform: lowercase;
  }
}

.weather-item__desc,
.header-day {
  display: none;
}
