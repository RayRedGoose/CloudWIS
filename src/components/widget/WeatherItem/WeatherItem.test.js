import React from 'react';
import { shallow } from 'enzyme';
import WeatherItem from './WeatherItem';

it('should match the snapshot with all data passed in correctly', () => {
  const mockProps = {
    key: 1,
    weekDay: '',
    typeAbbr: 'temp',
    itemType: 'Clouds',
    blockType: 'day',
    number: 15
  }

  const item = shallow(
    <WeatherItem {...mockProps} />
  );

  expect(item).toMatchSnapshot();
});
