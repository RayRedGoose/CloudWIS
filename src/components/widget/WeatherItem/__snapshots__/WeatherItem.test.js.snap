// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot with all data passed in correctly 1`] = `
<section
  className="weather-item weather-item__Clouds"
>
  <p
    className="header-day"
  />
  <p
    className="number temp"
  >
     
    15
  </p>
  <p>
    Clouds
  </p>
</section>
`;
